### PR TITLE
resolve ranking and issue 65

### PR DESCRIPTION
**Pull request by HoWeiChin**
To avoid propagating many changes across the codebase, I implement a isBefore method in the ParserUtil class.

In short, the isBefore method achieves the same result as the before method in the Date class. But, isBefore method requires checking of the dateFormat.

This avoids the drastic changes as compared to the previous PR which I made, to resolve the ranking issue and issue 65.

### CODE REVIEW

File: src/main/java/seedu/system/logic/parser/Parser.java [hunk 1]
```diff
@@ -12,5 +12,5 @@
      * Parses {@code userInput} into a command and returns it.
      * @throws ParseException if {@code userInput} does not conform the expected format
      */
-    T parse(String userInput) throws ParseException;
+    T parse(String userInput) throws ParseException, java.text.ParseException;
```
Comment (ooimingsheng) [inline src/main/java/seedu/system/logic/parser/Parser.java]: whats the difference between parse exception and java.text.ParseException

File: src/main/java/seedu/system/model/person/CustomDate.java [hunk 2]
```diff
@@ -38,39 +38,16 @@ public static boolean isValidDate(String date) {
...
-            if (dayOne > dayTwo) {
-                return false;
-            }
-            return true;
+    public Date getDate() {
+        try {
+            return new SimpleDateFormat(DATE_FORMAT).parse(date);
+        } catch (ParseException e) {
+            return null;
         }
```
Comment (ooimingsheng) [inline src/main/java/seedu/system/model/person/CustomDate.java]: getDate() should return dateObj;

the parsing should be done in the constructor

this.dateObj = SimpleDateFormat(DATE_FORMAT).parse(date);

not in getDate()

File: src/main/java/seedu/system/logic/parser/outofsession/FindPersonCommandParser.java [hunk 3]
```diff
@@ -24,10 +24,9 @@ public FindPersonCommand parse(String args) throws ParseException {
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindPersonCommand.MESSAGE_USAGE));
-        }
```
Comment (ooimingsheng) [inline src/main/java/seedu/system/logic/parser/outofsession/FindPersonCommandParser.java]: remove empty line please

**Comment by HoWeiChin** [comment]
The latest commit resolves issues related to displaying ranking results, addPerson and editPerson.

**Pull request merged**